### PR TITLE
Nostr improvements fixes

### PR DESCRIPTION
**Pull request by ktecho**

### CODE REVIEW

File: web/src/lib/components/nostr/Button.svelte [hunk 1]
```diff
@@ -29,37 +29,91 @@
 
         message += '\n' + pmURL;
 
-        pool.connectAndSendMessage({
-            message,
-            nostrRoomId,
-        });
+        document.getElementById('nostrTextToSend').value = message;
```
Comment (ibz) [inline web/src/lib/components/nostr/Button.svelte:32]: Using `getElementById` works, but is a bit ugly to do in svelte. A nicer way would be declaring `message` outside the function and using `bind:value={message}` and `bind:checked={textConfirmationVisible}`.

File: web/src/lib/components/nostr/Button.svelte [hunk 2]
```diff
@@ -29,37 +29,91 @@
...
+        document.getElementById('nostrTextConfirmation').checked = true;
+    }
+
+    async function postToNostr() {
+        let message = document.getElementById('nostrTextToSend').value || '';
```
Comment (ibz) [inline web/src/lib/components/nostr/Button.svelte:37]: No `getElementById` also if you use the above suggestion.

File: web/src/lib/components/nostr/Button.svelte [hunk 3]
```diff
@@ -29,37 +29,91 @@
...
+
+        if (message && message !== '') {
+            pool.connectAndSendMessage({
+                message,
+                nostrRoomId,
+            });
+
+            await new Promise(resolve => setTimeout(resolve, 1000));
```
Comment (ibz) [inline web/src/lib/components/nostr/Button.svelte:45]: I saw you added a utility function for this, no? Better use that.

File: web/src/lib/components/settings/Nostr.svelte [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+<div class="mt-8">
+    <p>Plebeian Market uses <b>Nostr</b> to support the following functionalities:</p>
+    <ul class="list-disc list-inside">
+        <li class="mt-3">Powering the Market Square and the Stall chat</li>
+    </ul>
+
+    <p class="mt-10">Features <i>coming soon</i>:</p>
+    <ul class="list-disc list-inside">
+        <li class="mt-3">Login with Nostr</li>
+        <li>Publish your Resumé to Nostr</li>
```
Comment (ibz) [inline web/src/lib/components/settings/Nostr.svelte:66]: "résumé"

File: web/src/lib/components/settings/Nostr.svelte [hunk 5]
```diff
@@ -0,0 +1,78 @@
+<script lang="ts">
+    import {user} from "$lib/stores";
+    import {getPublicKey} from "nostr-tools";
+    import {onMount} from "svelte";
+    import {hasExtension} from "$lib/nostr/utils";
+
+    let nostr_public_key = getPublicKey($user.nostr_private_key);
+    let nostrExtensionEnabled: boolean;
+
+    onMount(async () => {
```
Comment (ibz) [inline web/src/lib/components/settings/Nostr.svelte]: Is this needed? Why not just `{#if hasExtension()}`?

File: web/src/lib/components/settings/Nostr.svelte [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    {:else}
+        <div class="alert alert-warning shadow-lg">
+            <div>
+                <svg xmlns="http://www.w3.org/2000/svg" class="stroke-current flex-shrink-0 h-6 w-6" fill="none" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-3L13.732 4c-.77-1.333-2.694-1.333-3.464 0L3.34 16c-.77 1.333.192 3 1.732 3z" /></svg>
+                <span>You don't have a Nostr extension</span>
+            </div>
+        </div>
+
+        <div class="mt-4 text-justify">
+            <p>You're using your Plebeian Market generated Nostr identity. We created this for you as we detected
```
Comment (ibz) [inline web/src/lib/components/settings/Nostr.svelte:35]: "so nobody knows your private key" not technically correct, also sounds a bit strange. Maybe "so nobody else has access to your private key"?

File: web/src/lib/components/nostr/Button.svelte (hunk 1 above)
Comment (ktecho) [reply web/src/lib/components/nostr/Button.svelte]: Of course. I don't know what I was thinking... Thanks!

**Pull request merged**